### PR TITLE
feat: Add Instagram Business/Creator integration

### PR DESCRIPTION
**Pull request by AlexanderHeffernan**
### Description

- **Purpose**: Implement comprehensive Instagram Business/Creator integration for Autohive, enabling content management, community engagement, and analytics tracking via Instagram Graph API v24.0.
- **Approach**: Built a complete integration using the Instagram Graph API with support for account information retrieval, content publishing (images, videos, reels, carousels, stories), comment moderation (read, reply, hide/unhide, delete), and insights/analytics. Includes robust testing, helper functions for media container polling, and comprehensive documentation.

**Type of change**
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Updates**

👉 Added Instagram Business/Creator integration with 8 actions: account management, content publishing, comment engagement, and insights retrieval

👉 Implemented media container polling system for reliable publishing with support for images, videos, reels, carousels, and stories

👉 Created comprehensive test suite with 20+ tests covering all actions and edge cases, plus updated main README with Instagram integration details

### Author(s) to check

- [x] Project and all contained modules builds successfully
- [x] Self-/dev-tested
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

**Review by AlexanderHeffernan — COMMENTED** [review]
## 🛡️ **Pull Request Review**
**Status:** Ready to merge
**Blockers:** 0
**Suggestions:** 2
**Reviewed:** January 05, 2026

This is a high-quality, comprehensive integration for Instagram Business/Creator. The code is modular, well-tested, and includes robust handling for async media creation.

✅ **Strengths**
- Robust media state polling in [`instagram/helpers.py`](https://github.com/Autohive-AI/integrations-autohive/blob/8ce4e7ca72db10f309e1173c3432500219b61879/instagram/helpers.py#L35) ensures reliable publishing by handling Instagram's async processing.
- Excellent test coverage in [`instagram/tests/test_instagram.py`](https://github.com/Autohive-AI/integrations-autohive/blob/8ce4e7ca72db10f309e1173c3432500219b61879/instagram/tests/test_instagram.py) mocking all major action pathways.

🟡 **Suggestions**
- Pagination support is limited. [`get_posts`](https://github.com/Autohive-AI/integrations-autohive/blob/8ce4e7ca72db10f309e1173c3432500219b61879/instagram/actions/media.py#L45) and [`get_comments`](https://github.com/Autohive-AI/integrations-autohive/blob/8ce4e7ca72db10f309e1173c3432500219b61879/instagram/actions/comments.py#L35) accept a `limit` but do not return a cursor for fetching subsequent pages, which limits access to older data.
- The API version is hardcoded in [`instagram/helpers.py:8`](https://github.com/Autohive-AI/integrations-autohive/blob/8ce4e7ca72db10f309e1173c3432500219b61879/instagram/helpers.py#L8). Moving this to `config.json` would make future version upgrades more visible and centralized.

🔴 **Blockers**
- None identified.

---
*This review was generated by Pull Request Reviewer, an AI agent operating on the Autohive platform.*

File: instagram/actions/insights.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
+"""
+Instagram Insights actions - Analytics for accounts and media.
+"""
+
+from autohive_integrations_sdk import ActionHandler, ActionResult, ExecutionContext
+from typing import Dict, Any, List
```
[the remark this answers is not in the document]
Comment (AlexanderHeffernan) [reply instagram/actions/insights.py]: Resolved.

File: instagram/instagram.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+"""
+
+from autohive_integrations_sdk import Integration
+import os
+
+config_path = os.path.join(os.path.dirname(__file__), "config.json")
+instagram = Integration.load(config_path)
+
+# Import actions to register handlers
+import actions
```
[the remark this answers is not in the document]
Comment (AlexanderHeffernan) [reply instagram/instagram.py]: Resolved

**Review by AlexanderHeffernan — COMMENTED** [review]
## 🛡️ **Pull Request Review**
**Status:** Ready to merge
**Blockers:** 0
**Suggestions:** 1
**Reviewed:** January 05, 2026

The addition of cursor-based pagination significantly improves the utility of this integration for accounts with large histories. The implementation looks clean and is well-documented.

✅ **Strengths**
- **Pagination Implemented:** Excellent work adding [`after_cursor`](https://github.com/Autohive-AI/integrations-autohive/blob/36abd3c1502e0fb1585f027c724570ce9a73459e/instagram/actions/media.py#L46) and [`next_cursor`](https://github.com/Autohive-AI/integrations-autohive/blob/36abd3c1502e0fb1585f027c724570ce9a73459e/instagram/actions/media.py#L86) support. This addresses the previous limitation regarding data access.
- **Code Quality:** The refactoring of imports and addition of `noqa` comments shows attention to detail and linting standards.

🟡 **Suggestions**
- The API version is still hardcoded in [`instagram/helpers.py:8`](https://github.com/Autohive-AI/integrations-autohive/blob/36abd3c1502e0fb1585f027c724570ce9a73459e/instagram/helpers.py#L8). Moving `v24.0` to `config.json` is still recommended for easier maintenance in the future.

🔴 **Blockers**
- None identified.

---
*This review was generated by Pull Request Reviewer, an AI agent operating on the Autohive platform.*

File: instagram/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+# Response includes: { "media": [...], "next_cursor": "abc123" }
+
+# Next page
+{ "limit": 25, "after_cursor": "abc123" }
+```
+
+## Rate Limits
+
+- **Content Publishing**: 100 posts per 24 hours (carousels count as 1)
+- **API Calls**: Scales with account reach — `4800 × impressions in last 24hrs`. For example, an account with 1,000 impressions gets 4.8M API calls/day. Minimum floor of 48,000 calls/day for small accounts.
```
Comment (TheRealAgentK) [inline instagram/README.md]: I assume this is a rule from IG?

File: instagram/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+## Limitations
+
+- Media must be hosted on a publicly accessible URL for publishing
+- Stories are only available for 24 hours; insights expire after 24hrs
+- Carousel posts require 2-10 items
+- Video processing may take time before publishing completes
+
+## API Version
+
+This integration uses Instagram Graph API **v24.0**.
```
Comment (TheRealAgentK) [inline instagram/README.md]: What's our upgrade strategy and requirement here?

Comment (TheRealAgentK) [reply instagram/README.md]: To be more precise - iirc all Meta Graph-APIs have a kind of expiry and sunsetting pathway - are we getting notifications and/or alerts for these kind of things?
